src/oscore/oscore_cose.c

```diff
@@ -0,0 +1,537 @@
+  { "X448", COSE_curve_X448 },
+  { "Ed25519", COSE_curve_Ed25519 },
+  { "Ed448", COSE_curve_Ed448 },
+  { "secp256k1", COSE_curve_secp256k1 },
+};
+
+const char*
+cose_get_curve_name(cose_curve_t id, char* buffer, size_t buflen) {
+  for (size_t i = 0; i < sizeof(curve_mapping)/sizeof(curve_mapping[0]); i++) {
+    if (id == curve_mapping[i].id) {
+      snprintf(buffer, buflen, "%s (%d)", curve_mapping[i].name, id);
+      return buffer;
+    }
+  }
+  snprintf(buffer, buflen, "curve Fix me (%d)", id);
+  return buffer;
+}
+
+cose_curve_t
+cose_get_curve_id(const char *name) {
+  for (size_t i = 0; i < sizeof(curve_mapping)/sizeof(curve_mapping[0]); i++) {
+    if (strcmp(name, curve_mapping[i].name) == 0)
+      return curve_mapping[i].id;
+  }
+  return 0;
+}
+
+struct cose_alg_desc {
+  const char *name;
+  cose_alg_t id;
+};
+
+static struct cose_alg_desc alg_mapping[] = {
+  { "ES256K", COSE_Algorithm_ES256K },
+  { "SHA-512", COSE_Algorithm_SHA_512 },
+  { "SHA-384", COSE_Algorithm_SHA_384 },
+  { "ES512", COSE_Algorithm_ES512 },
+  { "ES384", COSE_Algorithm_ES384 },
+  { "ECDH-SS + HKDF-256", COSE_Algorithm_ECDH_SS_HKDF_256 },
+  { "SHA-512/256", COSE_Algorithm_SHA_512_256 },
+  { "SHA-256", COSE_Algorithm_SHA_256_256 },
+  { "SHA-256/64", COSE_Algorithm_SHA_256_64 },
+  { "SHA-1", COSE_Algorithm_SHA_1 },
+  { "direct+HKDF-SHA-512", COSE_Algorithm_HKDF_SHA_512 },
+  { "direct+HKDF-SHA-256", COSE_Algorithm_HKDF_SHA_256 },
+  { "EdDSA", COSE_Algorithm_EdDSA },
+  { "ES256", COSE_Algorithm_ES256 },
+  { "HMAC 256/64", COSE_Algorithm_HMAC256_64 },
+  { "HMAC 256/256", COSE_Algorithm_HMAC256_256 },
+  { "HMAC 384/384", COSE_Algorithm_HMAC384_384 },
+  { "HMAC 512/512", COSE_Algorithm_HMAC512_512 },
+  { "AES-CCM-16-64-128", COSE_Algorithm_AES_CCM_16_64_128 },
+  { "AES-CCM-16-64-256", COSE_Algorithm_AES_CCM_16_64_256 },
+  { "AES-CCM-64-64-128", COSE_Algorithm_AES_CCM_64_64_128 },
+  { "AES-CCM-64-64-256", COSE_Algorithm_AES_CCM_64_64_256 },
+  { "ChaCha20/Poly1305", COSE_Algorithm_ChaCha20_P1035 },
+  { "AES-CCM-16-128-128", COSE_Algorithm_AES_CCM_16_128_128 },
+  { "AES-CCM-16-128-256", COSE_Algorithm_AES_CCM_16_128_256 },
+  { "AES-CCM-64-128-128", COSE_Algorithm_AES_CCM_64_128_128 },
+  { "AES-CCM-64-128-256", COSE_Algorithm_AES_CCM_64_128_256 },
+};
+
+const char*
+cose_get_alg_name(cose_alg_t id, char* buffer, size_t buflen) {
+  for (size_t i = 0; i < sizeof(alg_mapping)/sizeof(alg_mapping[0]); i++) {
+    if (id == alg_mapping[i].id) {
+      snprintf(buffer, buflen, "%s (%d)", alg_mapping[i].name, id);
+      return buffer;
+    }
+  }
+  snprintf(buffer, buflen, "alg Fix me (%d)", id);
+  return buffer;
+}
+
+cose_alg_t
+cose_get_alg_id(const char *name) {
+  for (size_t i = 0; i < sizeof(alg_mapping)/sizeof(alg_mapping[0]); i++) {
+    if (strcmp(name, alg_mapping[i].name) == 0)
+      return alg_mapping[i].id;
+  }
+  return 0;
+}
+
+/* return tag length belonging to cose algorithm */
+int
+cose_tag_len(int cose_alg) {
```

Comment:
The parameter type differs from that in the header file.

Reply:
Corrected, as well as in a few other similar cases.